crates/bevy_ecs/src/query/state.rs

```diff
@@ -199,14 +199,6 @@ impl<D: QueryData, F: QueryFilter> QueryState<D, F> {
             }
         }
 
-        if state.component_access.access().has_write_all_resources() {
```

Comment:
It might be worth adding a check like

```rust
debug_assert!(!state.component_access.access().has_any_resource_write());
```

so that anyone who tries it gets an error.